src/Turnip/Eval/Metatables.hs

```diff
@@ -24,3 +24,15 @@ getMetaFunction fstr v = do
                 _                  -> return Nothing
         Nothing -> return Nothing
 
+-- |This is a special function designed to extract the special case where
+-- 
+getMetaIndexTable :: Value -> LuaM (Maybe TableRef)
```

Comment:
Should this whole thing get special treatment right here? Say automatic `metaFieldRef`?